studio/src/rte/rte-field.js

```diff
@@ -0,0 +1,746 @@
+                    align-content: center;
+                }
+
+                a.accent,
+                a.primary-outline,
+                a.secondary,
+                a.secondary-outline {
+                    height: 32px;
+                    text-decoration: none;
+                    padding: 0 14px;
+                    border-radius: 16px;
+                }
+
+                a.primary-link,
+                a.secondary-link {
+                    height: initial;
+                    padding: 0 4px;
+                }
+
+                a.accent {
+                    background-color: var(--spectrum-global-color-blue-500);
+                    color: var(--spectrum-global-color-gray-50);
+                }
+
+                a.primary-outline {
+                    background-color: initial;
+                    border: 2px solid var(--spectrum-global-color-gray-900);
+                    color: var(--spectrum-global-color-gray-900);
+                }
+
+                a.secondary {
+                    color: var(--spectrum-global-color-gray-900);
+                    background-color: var(--spectrum-global-color-gray-200);
+                }
+
+                a.secondary-outline {
+                    background-color: initial;
+                    border: 2px solid var(--spectrum-global-color-gray-300);
+                    color: var(--spectrum-global-color-gray-900);
+                }
+
+                a.primary-link {
+                    background-color: initial;
+                    color: var(--spectrum-blue-900);
+                }
+
+                a.secondary-link {
+                    border: none;
+                    color: var(--spectrum-gray-800);
+                }
+
+                .ProseMirror span[is='inline-price'],
+                .ProseMirror a {
+                    cursor: default;
+                    display: inline-block;
+                    vertical-align: baseline;
+                    white-space: nowrap;
+                    margin: 0 1px;
+                }
+
+                .price.price-strikethrough {
+                    text-decoration: line-through;
+                }
+
+                div.ProseMirror-focused
+                    span[is='inline-price'].ProseMirror-selectednode,
+                div.ProseMirror-focused a.ProseMirror-selectednode {
+                    outline: 2px dashed var(--spectrum-global-color-blue-500);
+                    outline-offset: 2px;
+                    border-radius: 16px;
+                }
+            `,
+            prosemirrorStyles,
+        ];
+    }
+
+    #boundHandlers;
+    #editorSchema;
+    editorView;
+    value = '';
+    #serializer;
+
+    constructor() {
+        super();
+        this.readOnly = false;
+        this.isLinkSelected = false;
+        this.priceSelected = false;
+        this.showLinkEditor = false;
+        this.inline = false;
+        this.link = false;
+        this.#boundHandlers = {
```

Comment:
is it complicated to abstract plugins? i'm thinking of 2 things:
- complexity & ramp ups for future additions, where we want to add things like see terms or stuff like this, 
- maintenance of rte-field that gathers lot of logicwhere rte-field file will be heavily modified. If your answer is yes i'm fine leaving this like that.

Reply:
@npeltier my understanding of Prosemirror is that, since it is highly customizable, you create your own RTE with it, but not necessarily a customizable RTE with it, as it would require you to re-expose many low level features of Prosemirror somehow.
The learning curve is IMO high and I wanted to have something that works for us in this first iteration.
I already have some ideas how to approach placeholders implementation, but who ever works on it, will need to spend some time to learn about Prosemirror.